safe_transaction_service/tokens/tasks.py

```diff
@@ -111,3 +113,33 @@ def fix_pool_tokens_task() -> Optional[int]:
         if number:
             logger.info("%d pool token names were fixed", number)
         return number
+
+
+@app.shared_task()
+@close_gevent_db_connection_decorator
+def update_token_info_from_token_list() -> int:
+    """
+    If there's at least one valid token list with at least 1 token, every token in the DB is marked as `not trusted`
+    and then every token on the is marked as `trusted`
+
+    :return: Number of tokens marked as `trusted`
+    """
+    tokens = []
+    for token_list in TokenList.objects.all():
+        tokens += token_list.get_tokens()
+
+    if not tokens:
+        return 0
+
+    # Make sure current chainId matches the one in the list
+    ethereum_network = get_ethereum_network()
+
+    token_addresses = [
+        fast_to_checksum_address(token["address"])
+        for token in tokens
+        if token["chainId"] == ethereum_network.value
+    ]
```

Comment:
I think it's better if we use a set here.

The reason being that the token lists can be potentially quite big. When we do a `filter(address__in=token_addresses)` I believe that the only way to get that query set would be that for every token _T_ we check if it's part of the of the token list with _N_ elements. So it'd iterate over a collection of _N_ (O_(T ** N)_) when we are just interested if it's part of the list or not (_O(T)_).

What do you think?

Reply:
I tested it with 100K addresses and query is instant, I thought about your solution but I think the current approach should work and be more optimal.

Also I took a look at:
- https://stackoverflow.com/questions/1009706/postgresql-max-number-of-parameters-in-in-clause
- https://stackoverflow.com/questions/4936731/maximum-length-of-an-sql-query

WDYT?

Reply:
Should be ok. I think that in the end it doesn't matter for how Django processes it. I took a look into:

https://github.com/django/django/blob/d8e1442ce2c56282785dd806e5c1147975e8c857/django/db/models/fields/related_lookups.py#L40-L62

And it seems that for this use-case it's serialised as a tuple so having a `list` or a `set` shouldn't matter.

But good to know! 😅

Reply:
Thanks for bringing this up